src/main/java/com/example/controllers/FilterController.java

```diff
@@ -0,0 +1,123 @@
+package com.example.controllers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Currency;
+
+@RequestMapping("/filter")
+@RestController
+public class FilterController {
+
+	@Autowired
+	CurrencyRepository cr;
+
+	// Just for testing
```

Comment:
Just a tip: try to use debugging instead printing as much as possible, because it becomes much more convenient if the project is bigger, it is possible to evaluate expressions on the fly etc.